examples/echo/echo_responder.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements a chip-echo-responder, for the
+ *      CHIP Echo Protocol.
+ *
+ *      The CHIP Echo Protocol implements two simple methods, in the
+ *      style of ICMP ECHO REQUEST and ECHO REPLY, in which a sent
+ *      payload is turned around by the responder and echoed back to
+ *      the originator.
+ *
+ */
+
+#include "common.h"
+
+#include <protocols/echo/Echo.h>
+#include <support/ErrorStr.h>
+
+// The EchoServer object.
+static Protocols::EchoServer gEchoServer;
+static SecureSessionMgr<Transport::UDP> gSessionManager;
+static SecurePairingUsingTestSecret gTestPairing;
+
+// Callback handler when a CHIP EchoRequest is received.
+static void HandleEchoRequestReceived(NodeId nodeId, System::PacketBuffer * payload)
+{
+    printf("Echo Request from node %lu, len=%u ... sending response.\n", nodeId, payload->DataLength());
```

Comment:
The README makes it sound like responses get sent....

Reply:
Yes, this callback allow the app to take more actions upon receiving the echo request, the echo response is sent by Echo protocol automatically. In this example, app just print out a message.